pycgm/prep.py

```diff
@@ -0,0 +1,695 @@
+    segment_dict['RShin'] = ['RTIB','RSH2','RSH3','RSH4']
+    segment_dict['LShin'] = ['LTIB','LSH2','LSH3','LSH4']
+    segment_dict['RFoot'] = ['RLFT1','RFT2','RMFT3','RLUP']
+    segment_dict['LFoot'] = ['LLFT1','LFT2','LMFT3','LLUP']
+    segment_dict['RHum'] = ['RMELB','RSHO','RUPA']
+    segment_dict['LHum'] = ['LMELB','LSHO','LUPA']
+    
+    return segment_dict
+
+def get_marker_location(pm, c):
+    """Finds the location of the missing marker in the world frame.
+
+    Parameters
+    ----------
+    pm : ndarray or list
+        Location of the missing marker in the cluster frame. 1x3 list or
+        numpy array.
+    c : 2darray or list
+        2d array or list indicating locations of other markers in the same
+        cluster and same frame as the missing marker.
+        Contains 3 1x3 lists or numpy arrays: `[X, Y, Z]`.
+
+    Returns
+    -------
+    pw : array
+        Location of the missing marker in the world frame. List of
+        3 elements `[X, Y, Z]`.
+    
+    Examples
+    --------
+    >>> from numpy import array, around
+    >>> pm = [-205.14696889756505, 258.35355899445926, 3.279423067505604]
+    >>> c = [array([ 325.82983398,  402.55450439, 1722.49816895]),
+    ...      array([ 304.39898682,  242.91339111, 1694.97497559]),
+    ...      array([ 197.8621521 ,  251.28889465, 1696.90197754])]
+    >>> around(get_marker_location(pm, c), 8) #doctest: +NORMALIZE_WHITESPACE
+    array([ 187.23396416, 407.91688108, 1720.71952837])
+    """
+    #Pm is the location of the missing marker in the cluster frame
+    # C = [origin,x_dir,y_dir] 
+    # create Tw_c is the cluster frame in the world frame
+    # find Pw, the missing marker in the world frame
+    
+    origin = c[0]
+    x_dir = c[1]
+    y_dir = c[2]
+    
+    x_vec = x_dir - origin
+    y_vec = y_dir - origin
+    x_hat = x_vec / np.linalg.norm(x_vec)
+    y_hat = y_vec / np.linalg.norm(y_vec)
+    z_vec = np.cross(x_hat,y_hat)
+    z_hat = z_vec / np.linalg.norm(z_vec)
+    
+    #Define the transformation matrix of the cluster in world space, World to Cluster
+    tw_c =np.matrix([[x_hat[0],y_hat[0],z_hat[0],origin[0]],
+                     [x_hat[1],y_hat[1],z_hat[1],origin[1]],
+                     [x_hat[2],y_hat[2],z_hat[2],origin[2]],
+                     [0       ,0       ,0       ,1        ]])
+            
+    #Define the transfomration matrix of the marker in cluster space, cluster to Marker
+    tc_m = np.matrix([[1,0,0,pm[0]],
+                      [0,1,0,pm[1]],
+                      [0,0,1,pm[2]],
+                      [0,0,0,1   ]])
+        
+    #Find Pw, the marker in world space
+    # Take the transform from world to cluster, then multiply cluster to marker
+    tw_m = tw_c * tc_m
+    
+    #The marker in the world frame
+    pw = [tw_m[0,3],tw_m[1,3],tw_m[2,3]]
+    
+    return pw
+
+def get_static_transform(p, c):
+    """Finds the location of the missing marker in the cluster frame.
+
```

Comment:
this seems to be missing some key explanations of what the parameters are.  It is also important to have the math